server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java

```diff
@@ -93,164 +98,213 @@
+            enqueueSomeMails(namePatternForSlice(2), emailCount);
+
+            clock.setInstant(IN_SLICE_3);
+            enqueueSomeMails(namePatternForSlice(3), emailCount);
+
+            clock.setInstant(IN_SLICE_5);
+            enqueueSomeMails(namePatternForSlice(5), emailCount);
+
+            clock.setInstant(IN_SLICE_7);
+            dequeueMails(5);
+            dequeueMails(5);
+            dequeueMails(3);
+
+            Stream<String> names = Iterators.toStream(mailQueue.browse())
+                .map(ManageableMailQueue.MailQueueItemView::getMail)
+                .map(Mail::getName);
+
+            assertThat(names)
+                .containsExactly("3-4", "3-5", "5-1", "5-2", "5-3", "5-4", "5-5");
+        }
+
+        private Function<Integer, String> namePatternForSlice(int sliceId) {
+            return i -> sliceId + "-" + i;
+        }
+
+        @Test
+        void mailQueueShouldBeInitializedWhenCreating(CassandraCluster cassandra) {
+            String name = "myQueue";
+            mailQueueFactory.createQueue(name);
+
+            boolean initialized = CassandraMailQueueViewTestFactory.isInitialized(cassandra.getConf(), MailQueueName.fromString(name));
+            assertThat(initialized).isTrue();
+        }
+
+        @Test
+        void enQueueShouldNotThrowOnMailNameWithNegativeHash() {
+            String negativehashedString = "this sting will have a negative hash"; //hash value: -1256871313
+
+            assertThatCode(() -> getMailQueue().enQueue(defaultMail().name(negativehashedString).build()))
+                .doesNotThrowAnyException();
+        }
+
+        @Disabled("JAMES-2614 RabbitMQMailQueueTest::concurrentEnqueueDequeueShouldNotFail is unstable." +
+            "The related test is disabled, and need to be re-enabled after investigation and a fix.")
+        @Test
+        @Override
+        public void concurrentEnqueueDequeueShouldNotFail() {
+
+        }
+
+        private void enqueueSomeMails(Function<Integer, String> namePattern, int emailCount) {
+            IntStream.rangeClosed(1, emailCount)
+                .forEach(Throwing.intConsumer(i -> enQueue(defaultMail()
+                    .name(namePattern.apply(i))
+                    .build())));
+        }
+
+        private void dequeueMails(int times) {
+            Flux.from(getManageableMailQueue()
+                .deQueue())
+                .take(times)
+                .flatMap(mailQueueItem -> Mono.fromCallable(() -> {
+                    mailQueueItem.done(true);
+                    return mailQueueItem;
+                }))
+                .blockLast();
+        }
     }
 
-    @Disabled("JAMES-2614 RabbitMQMailQueueTest::concurrentEnqueueDequeueShouldNotFail is unstable." +
-        "The related test is disabled, and need to be re-enabled after investigation and a fix.")
-    @Test
-    @Override
-    public void concurrentEnqueueDequeueShouldNotFail() {
+    @Nested
+    class MailQueueSizeMetricsDisabled {
```

Comment:
We don't implement  ManageableMailQueueContract, MailQueueMetricContract. 
Can we still implement MailQueueContract ?

Reply:
Is this really necessary? This class is just to test that we disabled the metric for mail queue size. The contracts are implemented on the other nested class, that is pretty much having the tests from before. 

Isn't that enough? Wouldn't it be redundant to add a contract here? Can you detail me more why we should add again a contract?

Reply:
Yes IMO playing these tests twice will be overkill (expensive test suite) for very little safety gains.